File: SdkTestAutomation.Sdk/Models/SdkResponse.cs
using Newtonsoft.Json;

namespace SdkTestAutomation.Sdk.Models;

public class SdkResponse<T>
{
    [JsonProperty("success")]
    public bool Success { get; set; }
    
    [JsonProperty("data")]
    public T Data { get; set; }
    
    [JsonProperty("content")]
    public string Content { get; set; } = string.Empty;
    
    [JsonProperty("errorMessage")]
    public string ErrorMessage { get; set; } = string.Empty;
    
    [JsonProperty("statusCode")]
    public int StatusCode { get; set; } = 200;
    
    public static SdkResponse<T> CreateSuccess(T data) => new() { Success = true, Data = data };
    public static SdkResponse<T> CreateError(string message, int statusCode = 500) => new() { Success = false, ErrorMessage = message, StatusCode = statusCode };
} 

File: SdkTestAutomation.CliWrappers/SdkTestAutomation.Go/sdk_response/sdk_response.go
package sdk_response

import "encoding/json"

// SdkResponse represents a standardized response from SDK operations
type SdkResponse struct {
	Success      bool        `json:"success"`
	Data         interface{} `json:"data,omitempty"`
	ErrorMessage *string     `json:"errorMessage,omitempty"`
}

// CreateSuccess creates a successful response with optional data
func CreateSuccess(data ...interface{}) *SdkResponse {
	response := &SdkResponse{
		Success: true,
	}
	
	if len(data) > 0 {
		response.Data = data[0]
	}
	
	return response
}

// CreateError creates an error response with the given message
func CreateError(statusCode int, message string) *SdkResponse {
	return &SdkResponse{
		Success:      false,
		ErrorMessage: &message,
	}
}

// ToJSON converts the response to JSON string
func (r *SdkResponse) ToJSON() (string, error) {
	bytes, err := json.Marshal(r)
	if err != nil {
		return "", err
	}
	return string(bytes), nil
} 